### ISSUE TITLE
Feature: Allow glob option `matchBase`

### ISSUE THREAD

**Issue by nottoseethesun** [igorshubovych/markdownlint-cli#65]
Currently, this seems necessary when, as is common for nodejs projects, there's a `README.md` in the top-level directory and some `.md` files in the `./src` tree:

```bash
markdownlint --ignore node_modules *.md && markdownlint --ignore node_modules **/*.md 
```

Instead if [glob](https://github.com/isaacs/node-glob) option [`matchBase`](https://github.com/isaacs/node-glob#basename-matching) could be set to `true`, then something like the following would suffice:

```bash
markdownlint --ignore node_modules --matchBase *.md 
```

**Comment by nschonni** [comment]
This usually works fine instead of the dual execution `markdownlint "**/*.md" -i "node_modules"`. Quoting is sometimes needed for globing to expand properly

**Comment by DavidAnson** [comment]
What @nschonni said. :) Here's an example where only violations in the project root and src directories are reported:

```text
C:\G\markdownlint-cli-issue-65>npm run mdl

> markdownlint-cli-issue-65@ mdl C:\G\markdownlint-cli-issue-65
> markdownlint --ignore node_modules **/*.md

bad.md: 1: MD047/single-trailing-newline Files should end with a single newline character
src/bad.md: 1: MD047/single-trailing-newline Files should end with a single newline character
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! markdownlint-cli-issue-65@ mdl: `markdownlint --ignore node_modules **/*.md`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the markdownlint-cli-issue-65@ mdl script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\David\AppData\Roaming\npm-cache\_logs\2019-10-12T03_17_33_326Z-debug.log

C:\G\markdownlint-cli-issue-65>npm ls
markdownlint-cli-issue-65@ C:\G\markdownlint-cli-issue-65
`-- markdownlint-cli@0.17.0
  +-- commander@2.9.0
  | `-- graceful-readlink@1.0.1
  +-- deep-extend@0.5.1
  +-- get-stdin@5.0.1
  +-- glob@7.1.4
  | +-- fs.realpath@1.0.0
  | +-- inflight@1.0.6
  | | +-- once@1.4.0 deduped
  | | `-- wrappy@1.0.2
  | +-- inherits@2.0.4
  | +-- minimatch@3.0.4 deduped
  | +-- once@1.4.0
  | | `-- wrappy@1.0.2 deduped
  | `-- path-is-absolute@1.0.1
  +-- js-yaml@3.13.1
  | +-- argparse@1.0.10
  | | `-- sprintf-js@1.0.3
  | `-- esprima@4.0.1
  +-- lodash.differencewith@4.5.0
  +-- lodash.flatten@4.4.0
  +-- markdownlint@0.15.0
  | `-- markdown-it@8.4.2
  |   +-- argparse@1.0.10 deduped
  |   +-- entities@1.1.2
  |   +-- linkify-it@2.2.0
  |   | `-- uc.micro@1.0.6 deduped
  |   +-- mdurl@1.0.1
  |   `-- uc.micro@1.0.6
  +-- minimatch@3.0.4
  | `-- brace-expansion@1.1.11
  |   +-- balanced-match@1.0.0
  |   `-- concat-map@0.0.1
  `-- rc@1.2.8
    +-- deep-extend@0.6.0
    +-- ini@1.3.5
    +-- minimist@1.2.0
    `-- strip-json-comments@2.0.1

C:\G\markdownlint-cli-issue-65>
```

**Comment by DavidAnson** [comment]
Please see also this section I added to the documentation: https://github.com/igorshubovych/markdownlint-cli#globbing

I will close this issue as it seems not to be a problem, but please re-open it with more information if it’s still not working as we expect.
